### PR TITLE
mark API tests that should also run on LDAP

### PR DESCRIPTION
**Pull request by individual-it**
## Description
- mark API tests that also should run on LDAP as such
- fix some steps, so that they fit with the .ldif file that we import into LDAP

## Motivation and Context
having better coverage for LDAP

## How Has This Been Tested?
:robot: 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [x] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/apiMain/comments.feature [hunk 1]
```diff
@@ -106,9 +106,9 @@ Feature: Comments
 		Given user "user0" has been created
 		And user "user1" has been created
 		And group "sharinggroup" has been created
```
Comment (phil-davis) [inline tests/acceptance/features/apiMain/comments.feature]: Need to create grp1 here
And a few other places.
CI fails:
```
--- Failed scenarios:

    /drone/src/tests/acceptance/features/apiMain/comments.feature:105
    /drone/src/tests/acceptance/features/apiMain/comments.feature:117
    /drone/src/tests/acceptance/features/apiMain/comments.feature:133
/drone/src/tests/acceptance/features/apiMain/comments.feature:208
```

File: tests/acceptance/features/apiSharing/deleteShare.feature [hunk 2]
```diff
@@ -8,8 +8,8 @@ Feature: sharing
 	Scenario Outline: Delete all group shares
 		Given using OCS API version "<ocs_api_version>"
 		And group "group1" has been created
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing/deleteShare.feature]: Need to create grp1 here.
```
--- Failed scenarios:

    /drone/src/tests/acceptance/features/apiSharing/deleteShare.feature:21
    /drone/src/tests/acceptance/features/apiSharing/deleteShare.feature:22
    /drone/src/tests/acceptance/features/apiSharing/deleteShare.feature:117
/drone/src/tests/acceptance/features/apiSharing/deleteShare.feature:118
```
and in a few places...

File: tests/acceptance/features/apiSharing/disableSharing.feature [hunk 3]
```diff
@@ -94,11 +94,11 @@ So that ownCloud users cannot share file or folder
 
 	Scenario Outline: user shares a file with user who is in his group when sharing outside the group has been restricted
 		Given using OCS API version "<ocs_api_version>"
-		And group "sharinggroup" has been created
-		And user "user0" has been added to group "sharinggroup"
-		And user "user1" has been added to group "sharinggroup"
+		And group "grp1" has been created
+		And user "user2" has been added to group "grp1"
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing/disableSharing.feature]: I don't think user2 exists.
```
--- Failed scenarios:

    /drone/src/tests/acceptance/features/apiSharing/disableSharing.feature:106
    /drone/src/tests/acceptance/features/apiSharing/disableSharing.feature:107
    /drone/src/tests/acceptance/features/apiSharing/disableSharing.feature:121
    /drone/src/tests/acceptance/features/apiSharing/disableSharing.feature:122
    /drone/src/tests/acceptance/features/apiSharing/disableSharing.feature:148
/drone/src/tests/acceptance/features/apiSharing/disableSharing.feature:149
```

File: tests/acceptance/features/apiSharing/gettingShares.feature [hunk 4]
```diff
@@ -66,13 +68,13 @@ Feature: sharing
...
 	Scenario Outline: User's own shares reshared to him don't appear when getting "shared with me" shares
 		Given using OCS API version "<ocs_api_version>"
-		And group "group0" has been created
-		And user "user0" has been added to group "group0"
-		And user "user0" has created a folder "/shared"
-		And user "user0" has moved file "/textfile0.txt" to "/shared/shared_file.txt"
-		And user "user0" has shared folder "/shared" with user "user1"
-		And user "user1" has shared folder "/shared" with group "group0"
-		When user "user0" sends HTTP method "GET" to OCS API endpoint "/apps/files_sharing/api/v1/shares?shared_with_me=true"
+		And group "grp1" has been created
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing/gettingShares.feature]: user2 does not exist yet

File: tests/acceptance/features/apiSharing/mergeShare.feature [hunk 5]
```diff
@@ -1,55 +1,55 @@
...
-		And user "user0" shares folder "/merge-test-outside-twogroups" with group "group2" using the sharing API
+		When user "user0" shares folder "/merge-test-outside-twogroups" with group "grp1" using the sharing API
+		And user "user0" shares folder "/merge-test-outside-twogroups" with group "grp4" using the sharing API
 		Then as "user1" the folder "/merge-test-outside-twogroups" should exist
 		And as "user1" the folder "/merge-test-outside-twogroups (2)" should not exist
 
 	Scenario: Merging shares for recipient when shared from outside with two groups with different permissions
-		Given group "group2" has been created
+		Given group "grp4" has been created
 		And user "user1" has been added to group "group2"
```
Comment (phil-davis) [inline tests/acceptance/features/apiSharing/mergeShare.feature]: s/group2/grp4

**Review by phil-davis — APPROVED** [review]

**Comment by individual-it** [comment]
backport in #32266

**Pull request merged**